### ISSUE TITLE
In the witness computation WASM code, `log(x**x)` prints garbage

### ISSUE THREAD

**Issue by bkomuves** [iden3/circom#158]
The `log()` pseudo-function is a useful debugging tool. However, it exhibits some strange behaviour when executed by the generated WASM witness computation code (this bug does not appear while compiling the circuit, and I couldn't test the generated CPP version because that just crashes with "Illegal instruction"...):

```
log(3**3);       // garbage
log(6**6);       // garbage

log(3**4);       // OK
log(4**3);       // OK

var x = 5;
log(x**x);       // garbage

var y = x**x;
log(y);          // but this is OK

log(x**(x+1));   // OK
log(x**(x-1));   // OK
```

**Comment by bkomuves** [comment]
(btw this was originally discovered by MiED on the ZKP MOOC discord, for the case `3**3`)

**Comment by clararod9** [comment]
Hi!

We have solved this bug in our last release v2.1.5 (it only affected to the code generated in WASM).  Let us know if the current solution is working for you.

Thank you for reporting the issue!
